model/task/task.go

```diff
@@ -4129,9 +4131,16 @@ func getTasksByVersionPipeline(versionID string, opts GetTasksByVersionOptions)
 		taskNamesAsRegex := strings.Join(opts.TaskNames, "|")
 		match[DisplayNameKey] = bson.M{"$regex": taskNamesAsRegex, "$options": "i"}
 	}
-	// Activated Time is needed to filter out generated tasks that have been generated but not yet activated
-	if !opts.IncludeEmptyActivation {
-		match[ActivatedTimeKey] = bson.M{"$ne": utility.ZeroTime}
+	if !opts.IsMainlineCommit {
```

Comment:
We should combine these if statements and update the comment to make it clearer why we handle these cases differently.